### ISSUE TITLE
Not able to declare defaults for sub-command args

### ISSUE THREAD

**Issue by itzg** [alexflint/go-arg#86]
There doesn't seem to be a way to set defaults on argument fields declared within a subcommand, such as the following code. Pre-populating the subcommand pointers defeats the switch cases after parsing args.

```go
type ServerCmd struct {
	Bind string
}

type ClientCmd struct {
	ServerAddress string
}

var args struct {
	Server *ServerCmd `arg:"subcommand:server"`
	Client *ClientCmd `arg:"subcommand:client"`
}

func main() {
	// if defaults are set like this
	args.Server = &ServerCmd{Bind:":7676"}
	args.Client = &ClientCmd{ServerAddress:"127.0.0.1:7676"}
	arg.MustParse(&args)

	switch {
	case args.Client != nil:
		// then this case always gets selected
		fmt.Println("Running client command")

	case args.Server != nil:
		// and never this
		fmt.Println("Running server command")
	}
}
```

Adding a default field tag, as mentioned in https://github.com/alexflint/go-arg/issues/68, could be an elegant way to solve this.

**Comment by alexflint** [comment]
Thanks for reporting this @itzg, and yes I agree this is a limitation of the current implementation. I agree that something like #68 is probably the way to go.

**Comment by alexflint** [comment]
Fixed in #91
